src/Microsoft.SqlTools.ServiceLayer/SqlAssessment/GenerateScriptOperation.cs

```diff
@@ -0,0 +1,164 @@
+        /// </exception>
+        public void Execute(TaskExecutionMode mode)
+        {
+            // This method is not re-enterable as a trade-off for avoiding
+            // overcomplicated multiple CancellationTokenSources management
+            if (cancellation != null)
+            {
+                throw new InvalidOperationException(SR.SqlAssessmentOperationExecuteCalledTwice);
+            }
+
+            cancellation = new CancellationTokenSource();
+
+            try
+            {
+                var scriptText = GenerateScript(Parameters, cancellation.Token);
+                if (scriptText != null)
+                {
+                    SqlTask?.AddScript(SqlTaskStatus.Succeeded, scriptText);
+                }
+            }
+            catch (Exception e)
+            {
+                ErrorMessage = e.Message;
+                Logger.Write(TraceEventType.Error, string.Format(
+                    CultureInfo.InvariantCulture,
+                    "SQL Assessment generate script operation failed with exception {0}",
+                    e.Message));
+
+                throw;
+            }
+            finally
+            {
+                var c = Interlocked.Exchange(ref cancellation, null);
+                c?.Dispose();
+            }
+        }
+
+        public void Cancel()
+        {
+            cancellation?.Cancel();
+        }
+
+        #region Helpers
+
+        internal static string GenerateScript(GenerateScriptParams generateScriptParams,
+            CancellationToken cancellationToken)
+        {
+            const string scriptPrologue =
+                @"IF (NOT EXISTS(SELECT * FROM INFORMATION_SCHEMA.TABLES WHERE TABLE_SCHEMA = 'dbo' AND  TABLE_NAME = 'AssessmentResult'))
+BEGIN
+    CREATE TABLE [dbo].[AssessmentResult](
+    [CheckName] [nvarchar](max) NOT NULL,
+    [CheckId] [nvarchar](max) NOT NULL,
+    [RulesetName] [nvarchar](max) NOT NULL,
+    [RulesetVersion] [nvarchar](max) NOT NULL,
+    [Severity] [nvarchar](max) NOT NULL,
+    [Message] [nvarchar](max) NOT NULL,
+    [TargetPath] [nvarchar](max) NOT NULL,
+    [TargetType] [nvarchar](max) NOT NULL,
+    [HelpLink] [nvarchar](max) NOT NULL,
+    [Timestamp] [datetimeoffset](7) NOT NULL
+    )
+END
+GO
+INSERT INTO [dbo].[AssessmentResult] ([CheckName],[CheckId],[RulesetName],[RulesetVersion],[Severity],[Message],[TargetPath],[TargetType],[HelpLink],[Timestamp])
+VALUES";
+
+            var sb = new StringBuilder();
+            if (generateScriptParams.Items != null)
+            {
+                sb.Append(scriptPrologue);
+                foreach (var item in generateScriptParams.Items)
+                {
+                    cancellationToken.ThrowIfCancellationRequested();
+
+                    if (item.Kind == AssessmentResultItemKind.Note)
+                    {
+                        sb.Append("\r\n('")
+                            .Append(Escape(item.DisplayName)).Append("','")
```

Comment:
Instead of a whole bunch of these append calls it'd be easier and faster just to just use string interpolation

Reply:
Replaced with string interpolation. Surprisingly, it was not faster, but much more readable.